src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -908,6 +908,12 @@ public async Task UpdateLicenseAsync(Guid organizationId, OrganizationLicense li
 
         public async Task DeleteAsync(Organization organization)
         {
+            var ssoConfig = await _ssoConfigRepository.GetByOrganizationIdAsync(organization.Id);
+            if (ssoConfig?.GetData()?.UseKeyConnector == true)
+            {
+                throw new BadRequestException("You cannot delete an Organization that is using Key Connector.");
+            }
```

Comment:
Lets put this in a private method to not add clutter to an already cluttered function